scene/2d/tile_map.cpp

```diff
@@ -3863,6 +3863,22 @@ PackedStringArray TileMap::get_configuration_warnings() const {
 		}
 	}
 
+	if (tile_set.is_valid() && tile_set->get_tile_shape() == TileSet::TILE_SHAPE_ISOMETRIC) {
+		bool warn = !is_y_sort_enabled();
+		if (!warn) {
+			for (int layer = 0; layer < (int)layers.size(); layer++) {
+				if (!layers[layer].y_sort_enabled) {
+					warn = true;
+					break;
+				}
+			}
+		}
+
+		if (warn) {
+			warnings.push_back(RTR("Isometric TileSet works best if TileMap and all its layers have Y-sort enabled."));
```

Comment:
I feel like a warning should be more stern than telling you what's best.

How about "Isometric TileSet will likely not look as intended without Y-sort enabled for the TileMap and all of its layers?"